Marlin/src/module/planner.cpp

```diff
@@ -798,8 +799,12 @@ void Planner::calculate_trapezoid_for_block(block_t * const block, const_float_t
            final_rate = CEIL(block->nominal_rate * exit_factor); // (steps per second)
 
   // Limit minimal step rate (Otherwise the timer will overflow.)
-  NOLESS(initial_rate, uint32_t(MINIMAL_STEP_RATE));
-  NOLESS(final_rate, uint32_t(MINIMAL_STEP_RATE));
+  NOLESS(initial_rate, MINIMAL_STEP_RATE);
+  NOLESS(final_rate, MINIMAL_STEP_RATE);
+  NOLESS(block->nominal_rate, MINIMAL_STEP_RATE);
+
+  //NOMORE(initial_rate, block->nominal_rate);
```

Comment:
It's commented out because it's presumed these situations are impossible, or they are handled later. If neither of these is the actual case, that means these lines are strictly needed and we should un-comment them.

Reply:
I don't think there is any reason to think we need the commented out line.